test/microbench/org/apache/cassandra/test/microbench/instance/ReadTest.java

```diff
@@ -140,20 +153,55 @@ public void setup() throws Throwable
     abstract Object[] writeArguments(long i);
 
     public void performWrite(String writeStatement, long ofs, long count) throws Throwable
+    {
+        if (threadCount == 1)
+            performWriteSerial(writeStatement, ofs, count);
+        else
+            performWriteThreads(writeStatement, ofs, count);
+    }
+
+    public void performWriteSerial(String writeStatement, long ofs, long count) throws Throwable
     {
         for (long i = ofs; i < ofs + count; ++i)
             execute(writeStatement, writeArguments(i));
     }
 
+    public void performWriteThreads(String writeStatement, long ofs, long count) throws Throwable
+    {
+        List<Future<Integer>> futures = new ArrayList<>();
+        for (long i = 0; i < count; ++i)
+        {
+            long pos = ofs + i;
+            futures.add(executorService.submit(() ->
+                                               {
+                                                   try
+                                                   {
+                                                       execute(writeStatement, writeArguments(pos));
+                                                       return 1;
+                                                   }
+                                                   catch (Throwable throwable)
+                                                   {
+                                                       throw Throwables.propagate(throwable);
+                                                   }
+                                               }));
+        }
+        long done = 0;
+        for (Future<Integer> f : futures)
+            done += f.get();
+        assert count == done;
+    }
```

Comment:
Were writes just taking too long and making it hard to iterate quickly on optimizations? (AFAICT, the benchmark only explicitly measures reads...)

Reply:
nit: There are some common bits of write logic between `WriteTest` and `ReadTest`, to the extent your might be able to factor out a "writer" class.

Reply:
It was helpful to run just the read test and get a basic idea of the write performance too (and also data size).

Refactored the benchmarks to extract the shared code.